recipes-ni/ni-base-system-image-tests/files/fs_permissions_diff.py

```diff
@@ -207,7 +207,7 @@ def strip_versions_from_path(path):
     stripped_path = re.sub(r"salt-\d+\.\d+_\d+_g[0-9a-f]+", "salt-SALTVERSION", path)
     stripped_path = re.sub(r"salt-\d+\.\d+\+\d+\.g[0-9a-f]+", "salt-SALTVERSION", stripped_path)
     # Strip everything that looks like an RT kernel version (so module and kernel version changes are ignored)
-    stripped_path = re.sub(r"\d+\.\d+\.\d+-rt\d+", "VERSION.VERSION.VERSION-rtVERSION", stripped_path)
+    stripped_path = re.sub(r"\d+\.\d+\.\d+-rt\d+(-next)?(-g[a-f0-9]+)?", "VERSION.VERSION.VERSION-rtVERSION", stripped_path)
```

Comment:
I think it's better to have two regexes - one for current kernel and one for -next kernel; and not strip the "-next" string in the next kernel one. This way there won't be collisions in stripped paths when the -next kernel is same version as current kernel.

Reply:
I have changed this now, and have tested it on both the current kernel and the next kernel.

Reply:
Since we have separate lines for -next and current, the next's regex should not have `(-next)` as optional and replaced string should also have -next. i.e., something like (not tested by me)

`re.sub(r"\d+\.\d+\.\d+-rt\d+(-next)(-g[a-f0-9]+)?", "VERSION.VERSION.VERSION-rtVERSION-next", stripped_path)`

Reply:
When we compare the current kernel files against the next kernel, this would show up as a diff where the only difference would be the addition of the word "-next", it would just cause noise in the test log.

Reply:
I hadn't realized that we can sometimes compare -next kernels with current kernels. Thanks for catching that